### PR TITLE
initial commit for using list field type

### LINKED ISSUE #2364: RecordStreamIface support for "list" field type
This is needed for multi-category classification (this is a subtask of #2335) and will be useful for the pass through encoders.

### PR DESCRIPTION
**Pull request by BoltzmannBrain**
Fixes #2364.

### CODE REVIEW

**Comment by BoltzmannBrain** [comment]
@scottpurdy please review

File: nupic/data/utils.py [hunk 1]
```diff
@@ -147,3 +147,16 @@ def serializeSdr(sdr):
...
+def parseStringList(s):
+  """Parse a string of space-separated numbers, returning a Python list."""
+  assert isinstance(s, basestring)
+  return [int(i) for i in s.split()]
+
+
+
+def stripList(listObj):
+  """Convert a list of numbers to a space-separated list."""
+  return string.replace(str(listObj)[1:-1], ",", "")
```
Comment (scottpurdy) [inline nupic/data/utils.py]: `return " ".join(listObj)`

**Comment by scottpurdy** [comment]
:+1: other than the one change.

File: nupic/data/utils.py (hunk 1 above)
Comment (BoltzmannBrain) [inline nupic/data/utils.py]: this doesn't work, maybe the docstring could be more clear

**Comment by BoltzmannBrain** [comment]
@scottpurdy should be good to merge now.

**Comment by scottpurdy** [comment]
:+1:

**Pull request merged**